onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1059,6 +1090,7 @@ def _infer_Slice(self, node):
                     new_sympy_shape[i] = self._new_symbolic_dim_from_output(node, 0, i)
         else:
             for i, s, e, t in zip(axes, starts, ends, steps):
+                e = handle_negative_index(e, new_sympy_shape[i])
```

Comment:
>e = handle_negative_index(e, new_sympy_shape[i]) [](http://example.com/codeflow?start=16&length=48)

This seems to cause a test failure in nuphar, with:
(Pdb) new_sympy_shape
[sequence, 1, 10]
(Pdb) starts
[-1]
(Pdb) ends
[-9223372036854775807]
(Pdb) axes
[0]
(Pdb) steps
[-1]
(Pdb) self.int_max_
2147483647

It seems when ends <= -self.int_max_ it should ignore the symbolic dim in handle_negative_index.

Reply:
Is this `test_loop_to_scan` in `onnxruntime_test_python_nuphar.py`? Is the slice `x[-1::-1]`?

Reply:
Yes, it's a negative step in that test case.